### PR TITLE
new: /shrug

### PR DESCRIPTION
**Pull request by Sedonya**
¯\_(ツ)_/¯

### CODE REVIEW

**Comment by fokkonaut** [comment]
Thanks, looks good so far!
Please make sure to use tab spacing though, you can see that it misaligns in the GitHub review.

**Comment by Sedonya** [comment]
image
bruh

File: src/game/server/ddracechat.cpp [hunk 1]
```diff
@@ -3246,3 +3246,13 @@ void CGameContext::ConRainbowNameVIP(IConsole::IResult *pResult, void *pUserData
...
+
+void CGameContext::ConShrug(IConsole::IResult *pResult, void *pUserData)
+{
+    CGameContext *pSelf = (CGameContext *)pUserData;
+    CCharacter *pChr = pSelf->GetPlayerChar(pResult->m_ClientID);
+    if(!pChr)
+        return;
+
+    pSelf->SendChat(pChr->GetPlayer()->GetCID(), pChr->GetPlayer()->m_LocalChat ? CHAT_LOCAL : CHAT_ALL, -1, "¯\\_(ツ)_/¯");
+}
```
Comment (fokkonaut) [inline src/game/server/ddracechat.cpp:3258]: Please add a newline :)

Comment (fokkonaut) [reply src/game/server/ddracechat.cpp:3258]: and make sure the spacing is tab everywhere

File: src/game/server/ddracechat.cpp [hunk 2]
```diff
@@ -3246,3 +3246,13 @@ void CGameContext::ConRainbowNameVIP(IConsole::IResult *pResult, void *pUserData
...
 }
+
+void CGameContext::ConShrug(IConsole::IResult *pResult, void *pUserData)
+{
+    CGameContext *pSelf = (CGameContext *)pUserData;
+    CCharacter *pChr = pSelf->GetPlayerChar(pResult->m_ClientID);
+    if(!pChr)
+        return;
+
+    pSelf->SendChat(pChr->GetPlayer()->GetCID(), pChr->GetPlayer()->m_LocalChat ? CHAT_LOCAL : CHAT_ALL, -1, "¯\\_(ツ)_/¯");
```
Comment (fokkonaut) [inline src/game/server/ddracechat.cpp:3257]: Please add the `pResult->m_ClientID` as parameter for `SpamProtectionClientID`.

File: src/game/server/ddracechat.cpp [hunk 3]
```diff
@@ -3246,3 +3246,13 @@ void CGameContext::ConRainbowNameVIP(IConsole::IResult *pResult, void *pUserData
 
 	pChr->RainbowName(!pChr->GetPlayer()->m_RainbowName, pResult->m_ClientID);
 }
+
+void CGameContext::ConShrug(IConsole::IResult *pResult, void *pUserData)
+{
+    CGameContext *pSelf = (CGameContext *)pUserData;
+    CCharacter *pChr = pSelf->GetPlayerChar(pResult->m_ClientID);
+    if(!pChr)
```
Comment (fokkonaut) [inline src/game/server/ddracechat.cpp:3254]: Not required to check for the character. I'd just check whether `m_apPlayers[pResult->m_ClientID]` exists, that's enough.

File: src/game/server/ddracechat.cpp (hunk 2 above)
Comment (fokkonaut) [reply src/game/server/ddracechat.cpp:3257]: And I think it's not required to check for local chat. Just send it to `CHAT_ALL` always.

**Comment by fokkonaut** [comment]
Ur not dumb! Thanks :)

**Pull request merged**